lib/features/cli/networking/download_logs.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+import 'dart:io';
+
+import '../../../core/async/event_synchronizer.dart';
+import '../../../core/logs/logger.dart';
+import '../../../core/networking/protocol/project_build.dart';
+
+/// Requests the logs of the last build of the [session] and writes them
+/// to the [outFile].
+Future<void> downloadLogs(
```

Comment:
This also applies to `download_binary.dart`

Would you not agree that we could separate the concerns here a little better? This downloads the logs **and** saves them. I would prefer the saving not to be handled in networking code.